server/models/user.py

```diff
@@ -51,62 +53,86 @@ def verify_password(plain_password, encrypted_password):
 
 
 async def get_user(name: str = None, userid: str = None):
-    if not name and userid:
+    """Builds an query and fetches user data."""
+
+    if not name and not userid:
         return None
 
-    query = ['SELECT * FROM `users`']
+    query = ["SELECT * FROM users WHERE"]
+    args = {}
 
+    # Small query builder.
     if name:
-        query.append(f" WHERE `safe_name`='{varka.toSafeName(name)}'")
-    elif userid:
-        query.append(f" WHERE `userid`='{id}'")
+        query.append("safe_name = :safe_name OR")
+        args.update({"safe_name": varka.toSafeName(name)})
```

Comment:
better to `args["safe_name"] = varka.toSafeName(name)` for single parameters

p.s. i only see this now since i've done it 238974238 times in the past ;)

Reply:
o yeah thought of that but update() looks cleaner, i can change it i guess

Reply:
yeaa i mean it's not going to be the performance bottleneck, just that the update method does a lot more work (creates dict, calls update method, iterates thru k/v and basically does args[k] = v)